source/DataElement/AMDSEventData.cpp

```diff
@@ -11,16 +13,46 @@ AMDSEventData::~AMDSEventData()
 {
 }
 
+AMDSEventData& AMDSEventData::operator =(AMDSEventData &sourceEventData) {
+	if (this != &sourceEventData) {
+		cloneData(&sourceEventData);
+	}
+
+	return *this;
+}
+
+// ======== implementation of AMDSLightWeightEventData  ===========
+
 AMDSLightWeightEventData::AMDSLightWeightEventData(QDateTime eventTime, QObject *parent) :
 	AMDSEventData(parent)
 {
 	eventTime_ = eventTime;
 }
 
+AMDSLightWeightEventData::AMDSLightWeightEventData(AMDSLightWeightEventData &eventData, QObject *parent)
+	:AMDSEventData(parent)
+{
+	(*this) = eventData;
+}
+
 AMDSLightWeightEventData::~AMDSLightWeightEventData()
 {
 }
 
+void AMDSLightWeightEventData::cloneData(AMDSEventData *sourceEventData)
+{
+	setEventTime(sourceEventData->eventTime());
+}
+
+//AMDSLightWeightEventData& AMDSLightWeightEventData::operator =(AMDSLightWeightEventData &sourceEventData)
```

Comment:
Extra commented lines could probably be removed since you have an implementation of `operator =` higher up in the file.